### ISSUE TITLE
Fix Unit Tests for Addons

### ISSUE THREAD

**Issue by nerrad** [eventespresso/event-espresso-core#318]
This error is showing up for a bunch of our nightly add-on test runs that happen against EE core master:

```
Catchable fatal error: Argument 1 passed to EE_Session_Mock::__construct() must be an instance of EventEspresso\core\services\cache\CacheStorageInterface, null given, called in /tmp/event-espresso-core/tests/mocks/core/EE_Session_Mock.core.php on line 62 and defined in /tmp/event-espresso-core/tests/mocks/core/EE_Session_Mock.core.php on line 81
```

See [here](https://travis-ci.org/eventespresso/eea-affiliatewp/jobs/368502951) for an example.  What's odd, is it only appears to happen on PHP5.6 builds BUT, those builds are also running against WordPress 4.2 (which is the triggering factor I'm assuming).

I only created this ticket in the EE core branch because I'm uncertain yet whether this just means we bump the version tested against in our add-ons to 4.5.0 (to match the new minimum version requirement of EE core) - and thus the minimum WP version requirement for the add-ons as well, or this means there's changes needed in core because of something exposed by these failing tests.  Either way it should be investigated.

Seth, I think this should at least be investigated fairly soon.  Brent could give an assessment of what needs done.

**Comment by sethshoultes** [comment]
@tn3rb can you please take a look?

**Comment by tn3rb** [comment]
I've looked at the code and it's all good as far as I can tell, but it does rely on a WordPress function `tests_add_filter()` that is part of their test setup.
It's possible that that function's use changes in WP 4.2. 
I feel that this issue is most likely a result of using that function and not a problem with our code in any way.

I personally like Darren's suggestion of bumping the version tested against in our add-ons to 4.5.0.

**Comment by sethshoultes** [comment]
@nerrad is that something you can do?

**Comment by nerrad** [comment]
Yes.  However, I'm _hoping_ bumping the tested against version of WP fixes, but if it does not we'll need to figure out what exactly is happening.

**Comment by nerrad** [comment]
Yes this works so I'm going to update all our add-on travis.yml files.  I'll just be doing them as commits against the add-on master branch

**Comment by nerrad** [comment]
Okay I've updated all the `.travis.yml` files for add-ons that have tests building on travis.  The next step is just to see how the next nightlies goes so I'll put this in the testing column for that.

**Comment by nerrad** [comment]
Looks like this did the trick.  Latest nightlies all pass (except for legit fails needing addressed).
